lib/stm32/f4/rcc.c

```diff
@@ -743,6 +727,8 @@ void rcc_clock_setup_hse_3v3(const struct rcc_clock_scale *clock)
 	rcc_wait_for_osc_ready(RCC_PLL);
 
 	/* Configure flash settings. */
+	flash_dcache_enable();
+	flash_icache_enable();
```

Comment:
hacky as it may have been, you've removed the option of running with these disabled.